keps/sig-apps/0003-kubernetes-application-api.md

```diff
@@ -0,0 +1,1108 @@
+    
+    // Ready is used to report the readiness of an Application. If an Application contains an ApplicationHealthCheck, 
+    // this field corresponds directly to the Readiness of the Pod generated from that health check. If the 
+    // Application does not contain an ApplicationHealthCheck, this field is set to Unknown.
+    Ready ConditionStatus json:`"ready,omitempty" protobuf:"bytes,3,opt,name=ready"`
+}
+
+// ApplicationSpec is the specification object for an Application. It specifies the Application's components, 
+// dependencies, and an optional health check. Application components can be retrieved using the Applications 
+// spec.selector to select the resources that match from the components GroupKind at the client's desired version.
+type ApplicationSpec struct {
+    // Type is the type of the application (e.g. WordPress, MySQL, Cassandra).
+    Type string `json:"type" protobuf:"bytes,1,name=type"`
+    
+    // Components is a map of the applications components to the kinds created by the application (e.g. Pods, Services, 
+    // Deployments, CRDS, etc). The keys of the Components map are the identifiers of the components. These may not 
+    // be the names of the components. For example, two instances of an Application of the same type may use components 
+    // with same identifier key, but the names of the components must differ in order to ensure uniqueness in the 
+    // namespace. The values in the Components map are corresponding GroupKinds of the Application's components. These 
+    // may be used, in conjunction with the Selector, to retrieve an Application's components at the client's desired 
+    // version.
+    Components map [string]metav1.GroupKind `json:"components,omitempty" protobuf:"bytes,2,opt,name=components"`
+    
+    // Dependencies is a list of other Applications that this Application depends on.
+    Dependencies []sting `json:"dependencies" protobuf:"bytes,3,opt,name=dependencies"`
+    
+    // Selector is a label query over kinds that created by the application. It must match the component objects' labels.
+    // More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#label-selectors
+    Selector *metav1.LabelSelector `json:"selector" protobuf:"bytes,4,opt,name=selector"`
+    
+    // HealthCheck is the name of an optional Deployment that provides the health status of the Application. When the 
+    // Deployment is healthy, the application is considered to be healthy.
+    HealthCheckDeployment string `json:"healthCheck" protobuf:"bytes,5,opt,name=healthCheck"`
+    
+    // Version is an optional version indicator for the application.
+    Version string `json:"version,omitempty" protobuf:"bytes,6,opt,name=version"`
+    
+    // Description is a brief string description of the application. 
+    Description string `json:"description,omitempty" protobuf:"bytes,7,opt,name=description"`
```

Comment:
Many of the fields in ApplicationSpec are intended to be human-readable strings. Consider grouping them under one object (e.g "Descriptor").

I think this would clarify the purpose of these fields, particularly for users who are only seeing the API object and not the "human readable" comments in this spec.